src/Common/src/Interop/Unix/System.Security.Cryptography.Native/Interop.Encode.cs

```diff
@@ -28,8 +28,18 @@ internal static byte[] OpenSslEncode<THandle>(GetEncodedSizeFunc<THandle> getSiz
             byte[] data = new byte[size];
 
             int size2 = encode(handle, data);
-            Debug.Assert(size == size2);
+            if (size2 < 1)
+            {
+                Debug.Fail(
+                    $"{nameof(OpenSslEncode)}: {nameof(getSize)} succeeded ({size}) and {nameof(encode)} failed ({size2})");
```

Comment:
Remind me why it is this Debug.Fail is valid?  According to the man page, methods like i2d_X509 that are passed to encode can fail and return a negative value.  But we don't believe that'll ever happen because...?

Reply:
It was already called on line 21, but with a null destination.  (and negative already threw).  Now it's just writing the bytes it pre-calculated.